### PR TITLE
fix tiled map object node layer & tileset from atlas supported

### PR DESCRIPTION
**Pull request by zvars**
tile properties in tileset editor：
image

### CODE REVIEW

**Review by zhakesi — APPROVED** [review]

File: cocos/tiledmap/tiled-types.ts [hunk 1]
```diff
@@ -369,6 +369,8 @@ export class TMXTilesetInfo {
 
     imageName: string | null = null;
 
+    imagexy: Vec2 | null = null;
```
Comment (pandamicro) [inline cocos/tiledmap/tiled-types.ts]:
```suggestion
    imageOffset: Vec2 | null = null;
```

File: cocos/tiledmap/tiled-types.ts [hunk 2]
```diff
@@ -384,9 +386,14 @@ export class TMXTilesetInfo {
         let gid = gid_ as unknown as number;
         gid &= TileFlag.FLIPPED_MASK;
         gid -= (this.firstGid as unknown as number);
-        const max_x = Math.floor((this.imageSize.width - this.margin * 2 + this.spacing) / (this._tileSize.width + this.spacing));
-        rect.x = Math.round((gid % max_x) * (this._tileSize.width + this.spacing) + this.margin);
-        rect.y = Math.round(Math.floor(gid / max_x) * (this._tileSize.height + this.spacing) + this.margin);
+        if (this.imagexy) {
+            rect.x = this.imagexy.x;
+            rect.y = this.imagexy.y;
```
Comment (pandamicro) [inline cocos/tiledmap/tiled-types.ts]:
```suggestion
        if (this.imageOffset) {
            rect.x = this.imageOffset.x;
            rect.y = this.imageOffset.y;
```

File: cocos/tiledmap/tiled-utils.ts [hunk 3]
```diff
@@ -96,7 +96,7 @@ export function fillTextureGrids (tileset: TMXTilesetInfo, texGrids: TiledTextur
 
         tileset.rectForGID(gid as unknown as GID, grid);
 
-        if (!spFrame || count > 1) {
+        if (!spFrame || count > 1 || tileset.imagexy) {
```
Comment (pandamicro) [inline cocos/tiledmap/tiled-utils.ts]:
```suggestion
        if (!spFrame || count > 1 || tileset.imageOffset) {
```

File: cocos/tiledmap/tmx-xml-parser.ts [hunk 4]
```diff
@@ -698,6 +698,14 @@ export class TMXMapInfo {
 
                     this.parentGID = (fgid + (parseInt(tile.getAttribute('id')!) || 0)) as any;
                     const tileImages = tile.getElementsByTagName('image');
+                    if (tile.hasAttribute('x') && tile.hasAttribute('y')) {
+                        tileset.imagexy = new Vec2(parseFloat(tile.getAttribute('x')!) || 0, parseFloat(tile.getAttribute('y')!) || 0);
```
Comment (pandamicro) [inline cocos/tiledmap/tmx-xml-parser.ts]:
```suggestion
                        tileset.imageOffset = new Vec2(parseFloat(tile.getAttribute('x')!) || 0, parseFloat(tile.getAttribute('y')!) || 0);
```

**Review by SantyWang — APPROVED** [review]

**Comment by pandamicro** [comment]
@cocos-robot run test cases

**Pull request merged**